### PR TITLE
[MINOR][CORE] Leverage modified Utils.classForName to reduce scalastyle off for Class.forName

### PR DESCRIPTION
**Pull request by HeartSaVioR**
## What changes were proposed in this pull request?

This patch modifies Utils.classForName to have optional parameters - initialize, noSparkClassLoader - to let callers of Class.forName with thread context classloader to use it instead. This helps to reduce scalastyle off for Class.forName.

## How was this patch tested?

Existing UTs.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -189,15 +189,23 @@ private[spark] object Utils extends Logging {
...
-  def classForName(className: String): Class[_] = {
-    Class.forName(className, true, getContextOrSparkClassLoader)
+  /**
+   * Preferred alternative to Class.forName(className), as well as
+   * Class.forName(className, initialize, loader) with current thread's ContextClassLoader.
+   */
+  def classForName(
+      className: String,
+      initialize: Boolean = true,
+      noSparkClassLoader: Boolean = false): Class[_] = {
```
Comment (HeartSaVioR) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: If we are pretty sure that we should always fail back to the class loader which loads Spark, we can remove this parameter and simplifies the calling. I just don't have deep knowledge to judge whether it is safety to do that, so here not-beauty named parameter comes.

File: core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala [hunk 2]
```diff
@@ -378,10 +378,8 @@ private[spark] object ClosureCleaner extends Logging {
     } else {
       logDebug(s"Cleaning lambda: ${lambdaFunc.get.getImplMethodName}")
 
-      // scalastyle:off classforname
-      val captClass = Class.forName(lambdaFunc.get.getCapturingClass.replace('/', '.'),
-        false, Thread.currentThread.getContextClassLoader)
-      // scalastyle:on classforname
+      val captClass = Utils.classForName(lambdaFunc.get.getCapturingClass.replace('/', '.'),
+        initialize = false, noSparkClassLoader = false)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala]: It looks like noSparkClassLoader should be true here?

**Review by srowen — CHANGES_REQUESTED** [review]
Seems fine as a cleanup. I wouldn't modify any behavior here.

File: core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala (hunk 2 above)
Comment (HeartSaVioR) [reply core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala]: Ah yes nice finding! Will fix.

**Comment by srowen** [comment]
Merged to master

**Comment by HeartSaVioR** [comment]
Thanks for reviewing and merging!

**Pull request closed without merging**